src/components/Channel/Channel.tsx

```diff
@@ -14,29 +14,208 @@ import DefaultEmojiIndex from 'emoji-mart/dist-modern/utils/emoji-index/nimble-e
 import DefaultEmojiPicker from 'emoji-mart/dist-modern/components/picker/nimble-picker';
 import debounce from 'lodash.debounce';
 import throttle from 'lodash.throttle';
-import PropTypes from 'prop-types';
-import { logChatPromiseExecution, Channel as StreamChannel } from 'stream-chat';
+import {
+  ChannelAPIResponse,
+  ChannelState,
+  Event,
+  logChatPromiseExecution,
+  Message,
+  MessageResponse,
+  SendMessageAPIResponse,
+  Channel as StreamChannel,
+  StreamChat,
+  UpdatedMessage,
+} from 'stream-chat';
 import { v4 as uuidv4 } from 'uuid';
 
-import { Attachment as DefaultAttachment } from '../Attachment';
+import {
+  channelReducer,
+  ChannelStateReducer,
+  initialState,
+} from './channelState';
 import { commonEmoji, defaultMinimalEmojis, emojiSetDef } from './emojiData';
-import { MessageSimple } from '../Message';
+
+import { useEditMessageHandler } from './hooks/useEditMessageHandler';
+import { useIsMounted } from './hooks/useIsMounted';
+import {
+  OnMentionAction,
+  useMentionsHandlers,
+} from './hooks/useMentionsHandlers';
+
+import { Attachment as DefaultAttachment } from '../Attachment';
 import {
   LoadingErrorIndicator as DefaultLoadingErrorIndicator,
   LoadingIndicator as DefaultLoadingIndicator,
 } from '../Loading';
+import { MessageSimple } from '../Message';
 
-import { channelReducer, initialState } from './channelState';
-import useMentionsHandlers from './hooks/useMentionsHandlers';
-import useEditMessageHandler from './hooks/useEditMessageHandler';
-import useIsMounted from './hooks/useIsMounted';
-
-import { ChannelContext, ChatContext, TranslationContext } from '../../context';
+import {
+  ChannelContextValue,
+  ChannelProvider,
+  MessageAttachments,
+  MessageToSend,
+  useChatContext,
+  useTranslationContext,
+} from '../../context';
```

Comment:
split this up into the individual imports for each `Channel/Chat/Translation` context so we can get rid of the `index.ts/js` files